.editorconfig

```diff
@@ -20,24 +17,13 @@ csharp_style_expression_bodied_constructors = false:silent
 csharp_style_expression_bodied_operators = false:silent
 csharp_indent_labels = one_less_than_current
 
-[*.sln]
-indent_style = tab
-
 # Xml project files
 [*.{config,csproj,njsproj,targets,vcxitems,vcxproj,vcxproj.filters}]
 indent_size = 2
-end_of_line = crlf
-insert_final_newline = false
```

Comment:
This will cause EditorConfig-enabled editors to insert a newline, where Visual Studio trims such newlines when modifying those types of files.

It will cause on and off final newline diffs depending on what tool was used to edit such files.

I strongly recommend to keep this one.
To reiterate, it's not about the VS templates only, but the IDE's editing behavior.

Reply:
> his will cause EditorConfig-enabled editors to insert a newline

No, this is not true. If no rule exists for this in editorconfig, then it's the editor's default behavior coming into play. For most editors it's based on auto-detection. The editorconfig spec defines not default for this type of rule, so having no rule at all is the most compatible setting.

> where Visual Studio trims such newlines when modifying those types of files.

VS doesn't trim. It handles mixed line endings transparently without modifying. Only when adding new lines it will choose a line ending style (from auto-detection or VS options settings).

> It will cause on and off final newline diffs depending on what tool was used to edit such files.

No. This allows you to edit those files on Linux with LF and on Windows they will be CRLF

Reply:
I think this remark is not about EOL, but about insert_final_newline (newline at end of file). It might be good to keep it as @JunielKatarn suggested.

Reply:
If the remark is about newlines at the end of files, then the statement is still incorrect. Visual Studio does not trim any CR or LF at the end of files when editing.

VS leaves everything as is and that's the behavior that we need. On the contrary, the existence of those rules has caused the PR's from @JunielKatarn to include such changes to the end of files which we neither want nor need (assuming that the current state is how we want everything to be).

It's generally important to understand that removing such rules doesn't mean that the opposite would happen. Removing rather means that the editor won't make changes in that regard. 

Also important: Visual Studio doesn't need any editorconfig rules at all. It works totally fine without any editorconfig in place. 
It's also very unusual to have those EOL and EOF rules when using VS. Never seen that before, it's in none of the templates, nobody does that because there's no need for this in case of VS. It does these things right just out of the box.

Reply:
I'd say we should leave `insert_final_newline = false` in. These are specific VS formats. VS never adds final new lines and removes them when found. It is safe and reasonable to leave this setting in the editorconfig, so other code editors won't accidentially insert final new lines when it is their default.